common/utilities/string/string-utilities.h

```diff
@@ -3,139 +3,189 @@
 
 #pragma once
 
+#include <algorithm>
 #include <string>
 
 namespace utilities {
-namespace string {
-
-
-template < typename T,
-           typename std::enable_if< std::is_arithmetic< T >::value >::
-               type * = nullptr >  // check if T is arithmetic during compile
-inline bool string_to_value( const std::string & str, T & result )
-{
-    // Converts string to value via a given argument with desire type
-    // Input:
-    //      - string of a number
-    //      - argument with T type
-    // returns:true if conversion succeeded
-    // NOTE for unsigned types:
-    // if T is unsinged and the given string represents a negative number (meaning it starts with
-    // '-'), then string_to_value returns false (rather than send the string to the std conversion
-    // function). The reason for that is, stoul will return return a number in that case rather than
-    // throwing an exception.
-
-    try
-    {
-        size_t last_char_idx = 0;
-        if( std::is_integral< T >::value )
+    namespace strings {
+        inline std::string hexify(unsigned char n)
```

Comment:
Too much code in this function... if it's only for a character then it should be ~3 lines, without reversing or even a while loop.